docs/ingress_analysis.md

```diff
@@ -0,0 +1,43 @@
+# Ingress Analysis
+
+## Supported Ingress Controllers:
+
+Ingress Controllers considered for ingress analysis:
+- Openshift ingress controller 
+- Nginx ingress controller
+
+#### Namespace Labels Supported:
+
+|Ingress Controller | Namespace labels supported|
+|-------------------|---------------------------|
+|Openshift ingress controller |namespace.name: openshift-ingress-operator|
+||kubernetes.io/metadata.name: openshift-ingress-operator|
+||network.openshift.io/policy-group: ingress|
+|Nginx ingress controller |tier: ingress|
+||kubernetes.io/metadata.name: ingress-nginx|
+||app.kubernetes.io/part-of: ingress-nginx|

```

Comment:
> * where are the labels taken from (links?)

the labels are common labels that are used in network-policy files (from seeing examples in multiple repos in github)
could not find official links with these specific common used labels.
only namespaces names of the ingress controllers are mentioned in the official docs (links were added to the section above of `supported ingress controllers`)
 
>     * what do "supported" mean? looking for all labels or one of them? where are they expected to be identified?

updated in the doc
 
>     * what about pod labels? are there any such labels we can consider?

podSelector labels to select ingress-controllers are not commonly used in network-policy rules.
if we have a network-policy in `default` namespace (for example) and want to specify an ingress connection to a pod in `default` we must use the namespaceSelector to specify the ingress-controller's namespace; so I think it is enough to check the namespaceSelector.
moreover; the namespace name of an official ingress-controller is const and common to all `ingress-controller pods` under it ; but an ingress-controller deployment name varies ; and we might not be able to guess it to include it in the output lines.

Reply:
> the labels are common labels that are used in network-policy files (from seeing examples in multiple repos in github) could not find official links with these specific common used labels. only namespaces names of the ingress controllers are mentioned in the official docs

Do you have the links to the repos that you found from which you referenced the labels?
This is the link from `nginx` of the `namespace` configured: https://github.com/nginxinc/kubernetes-ingress/blob/main/deployments/common/ns-and-sa.yaml 
It seems the namespace `nginx-ingress` does not have official labels added to it when installed, so I don't think we can rely on labels that could be added by random examples on certain repos.. we can add an option to add those configuration details for considering them in the analysis..(to consider on a separate issue, not in this PR.) Tthe label `kubernetes.io/metadata.name: ingress-nginx` can be considered, since it is automatically added to any namespace.

Official pod labels can be inferred from the installed deployment: https://github.com/nginxinc/kubernetes-ingress/blob/main/deployments/deployment/nginx-ingress.yaml 

Can you find also official YAML configurations for openshift ingress controller? 
We should try to be consistent with official configurations, and also add links to those we rely on in the documentation.

Reply:
found this for openshift :
https://github.com/openshift/cluster-ingress-operator/blob/f9dd81ab522f72233e2608f5e57a43e79a5079b5/manifests/00-namespace.yaml#L10

I don't see the label that was used in the example we have `network.openshift.io/policy-group: ingress` within it.
so I will keep for now only the official labels , and add this one `network.openshift.io/policy-group: ingress` in another PR.

I suggest adding pods labels in another PR too, since I see in the pod labels of `nginx` only name label, but didn't see any consideration of the deployment name of an ingress-controller in the official docs

Reply:
see also: 
https://github.com/openshift/cluster-ingress-operator/blob/f9dd81ab522f72233e2608f5e57a43e79a5079b5/pkg/manifests/assets/router/namespace.yaml#L13

https://github.com/openshift/cluster-ingress-operator/blob/f9dd81ab522f72233e2608f5e57a43e79a5079b5/pkg/operator/controller/ingress/namespace.go#L91

Reply:
updated doc with the additional openshift controller namespace; wip code update too